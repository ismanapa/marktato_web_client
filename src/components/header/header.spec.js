import React from 'react';
import { mount, shallow } from 'enzyme';
import Header from './header';

describe('Header component', () => {
  it('Should match snapshot', () => {
    const title = 'Expected title';

    const wrapper = shallow(<Header title={title}/>);

    expect(wrapper).toMatchSnapshot();
  });
  it('Should have specific title', () => {
    const title = 'Expected title';

    const wrapper = mount(<Header title={title}/>);

    expect(wrapper.find('Typography').first().contains(title)).toBeTruthy();
  });
});
