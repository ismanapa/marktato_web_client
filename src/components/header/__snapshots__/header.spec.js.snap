// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="Expected title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Typography)
        color="inherit"
        variant="h6"
      >
        Expected title
      </WithStyles(Typography)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Expected title",
        "color": "inherit",
        "variant": "h6",
      },
      "ref": null,
      "rendered": "Expected title",
      "type": [Function],
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Typography)
          color="inherit"
          variant="h6"
        >
          Expected title
        </WithStyles(Typography)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Expected title",
          "color": "inherit",
          "variant": "h6",
        },
        "ref": null,
        "rendered": "Expected title",
        "type": [Function],
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
